### DISCUSSION TITLE
development setup howto

### DISCUSSION THREAD

**Discussion by ferkulat** [Ericsson/codechecker#3468] [Q&A]
Hi there,

I wanted to debug something in codechecker and probably contribute a fix.
I am not a python dev.
But I thought: "Its python. Can not be that hard." 

... a few hours later ...

Tried a lot of things until I discovered the Makefile and ran 
- `make package`,
- `make venv_dev`,
- `make pip_dev_deps` in different order several times.

Managed to tell pycharm using the created `venv_dev`

Even `make test` ran and failed with one failing test. It is still success!

Now debugging `venv_dev/bin/CodeChecker` and get an exception:
~~~~
[Errno 2] No such file or directory: '/home/homedir/PycharmProjects/codechecker/venv_dev/share/codechecker/config/commands.json'
~~~~
Running `python scripts/build/create_commands.py -v -b venv_dev` shows nothing and also creates no `commands.json`.

I feel dumb at that moment.
I would love to have a step by step "How to setup a development environment until I can  debug Codechecker for dummies".

Thank you in advance

**Comment by SevHub** [comment]
Same problem here. I (probably) found a bug in CodeChecker, and would love to fix it, or at least dig deeper. But the only way I can get CodeChecker to run is via the pip installation. And when I change some files in the library the lib/site-packages folder it has no effect, as if these files were ignored... Maybe because on Windows the whole thing runs as a compiled exe?
